src/ut/alarm_test.cpp

```diff
@@ -162,6 +167,305 @@ MATCHER_P(VoidPointeeEqualsStr, value, "")
+    EXPECT_CALL(_mz, zmq_recv(_,_,_,_)).Times(1).WillOnce(Return(0));
+    
+    EXPECT_CALL(_mz, zmq_send(_,VoidPointeeEqualsStr("issue-alarm"),_,_)).Times(1).WillOnce(Return(0));
+    EXPECT_CALL(_mz, zmq_send(_,VoidPointeeEqualsStr(issuer),_,_)).Times(1).WillOnce(Return(0));
+    EXPECT_CALL(_mz, zmq_send(_,VoidPointeeEqualsStr("9997.4"),_,_)).Times(1).WillOnce(Return(0));
+    EXPECT_CALL(_mz, zmq_recv(_,_,_,_)).Times(1).WillOnce(Return(0));
+  }
+  // Raises an alarm twice using different states each time. We check above for
+  // two ZMQ messages, the second containing the updated raised state.
+  _multi_state_alarm.set_critical();
+  _multi_state_alarm.set_major();
+  // Causes the test thread to wait to ensure the zmq messages get sent before
+  // the test terminates.
+  for (unsigned int i = 0; i < 2; i++)
+  {
+    _mz.call_complete(ZmqInterface::ZMQ_RECV, 5);
+  }
+}
+
+// Raises a MultiStateAlarm at two of its possible states and then clears it. We
+// expect three ZMQ messages to be sent to the Alarm Agent notifying it of each
+// state change.
+TEST_F(AlarmTest, MultiStateAlarmClearing)
+{
+  {
+    InSequence s;
+    EXPECT_CALL(_mz, zmq_send(_,VoidPointeeEqualsStr("issue-alarm"),_,_)).Times(1).WillOnce(Return(0));
+    EXPECT_CALL(_mz, zmq_send(_,VoidPointeeEqualsStr(issuer),_,_)).Times(1).WillOnce(Return(0));
+    EXPECT_CALL(_mz, zmq_send(_,VoidPointeeEqualsStr("9997.6"),_,_)).Times(1).WillOnce(Return(0));
+    EXPECT_CALL(_mz, zmq_recv(_,_,_,_)).Times(1).WillOnce(Return(0));
+    
+    EXPECT_CALL(_mz, zmq_send(_,VoidPointeeEqualsStr("issue-alarm"),_,_)).Times(1).WillOnce(Return(0));
+    EXPECT_CALL(_mz, zmq_send(_,VoidPointeeEqualsStr(issuer),_,_)).Times(1).WillOnce(Return(0));
+    EXPECT_CALL(_mz, zmq_send(_,VoidPointeeEqualsStr("9997.5"),_,_)).Times(1).WillOnce(Return(0));
+    EXPECT_CALL(_mz, zmq_recv(_,_,_,_)).Times(1).WillOnce(Return(0));
+    
+    EXPECT_CALL(_mz, zmq_send(_,VoidPointeeEqualsStr("issue-alarm"),_,_)).Times(1).WillOnce(Return(0));
+    EXPECT_CALL(_mz, zmq_send(_,VoidPointeeEqualsStr(issuer),_,_)).Times(1).WillOnce(Return(0));
+    EXPECT_CALL(_mz, zmq_send(_,VoidPointeeEqualsStr("9997.1"),_,_)).Times(1).WillOnce(Return(0));
+    EXPECT_CALL(_mz, zmq_recv(_,_,_,_)).Times(1).WillOnce(Return(0));
+  }
+  // Raises a MultiStateAlarm with two of its possible states and then clears
+  // it.
+  _multi_state_alarm.set_warning();
+  _multi_state_alarm.set_minor();
+  _multi_state_alarm.clear();
+  
+  // Causes the test thread to wait to ensure the zmq messages get sent before
+  // the test terminates.
+  for (unsigned int i = 0; i < 3; i++)
+  {
+    _mz.call_complete(ZmqInterface::ZMQ_RECV, 5);
+  }
+}
+
+// Raises an Alarm (single state) and then simulates time moving forward by
+// thirty seconds. We should expect three ZMQ messages: the first caused by the
+// alarm being raised, the second and third caused by the function to re-send
+// alarms every 30 seconds.
```

Comment:
Ah, I think I got that wrong. In fact, it's two messages because there are two alarms. I'd explain that here rather than in the middle of the test.